lang/list.go

```diff
@@ -162,7 +162,52 @@ var (
 		blockCommentStart: ">>COMMENT",
 		blockCommentEnd:   "\nCOMMENT",
 	}
-	// TODO scala, erlang, dart, racket, Elixir
+	erlangGen = baseLang{
+		name:              "Erlang",
+		slug:              "erlang",
+		shortName:         "erl",
+		extension:         ".erl",
+		lineComment:       "%",
+		blockCommentStart: "%",
+		blockCommentEnd:   "%",
```

Comment:
I think Erlang doest not support block comment? maybe we can use something like:
```erlang
-ifdef(comment).
  Something to comment
  You can add text or
  function(Declaration) ->
     ...
  Which will removed from file
-endif.
```

Ref: https://stackoverflow.com/questions/25322269/why-is-there-no-multi-line-commenting-support-in-erlang

Reply:
I don't think so… it's a little bit tricky:

(run `erlc hello.erl`)

```erl
-module(hello).
-export([world/0]).

-ifdef(comment).
  Comments.
-endif.

world() ->
    io:format("Hello, World!~n").
```

This ↓ will cause an compile error: `unterminated '-ifdef'`

```erl
-ifdef(comment).
  Comments
-endif.
```

But this ↓ will not:

```erl
-ifdef(comment).
  Comments.
-endif.
```

There still seems to be syntax checking, notice the missing '.' in the first example.

---

(I don't know what `blockCommentStart` and `blockCommentEnd` affects in the code)

Erlang is an old language, so how about letting it go? A user who actually uses it knows how to fix the error in generated code.

(I just tried Erlang for about five minutes maybe ( >\_< ), so... not really familiar with it )

Refs:

- <https://stackoverflow.com/questions/1541859/string-delimiting-in-erlang>
- <https://www.oreilly.com/library/view/programming-erlang-2nd/9781941222454/f_0070.html>
- <http://erlang.org/pipermail/erlang-questions/2010-December/055118.html>

Reply:
I think a better way is to make a more robust logic to parse temp here

```
{{ .BlockCommentStart }}
{{ block "title" . }}{{ .Question.QuestionFrontendId }}. {{ .Question.GetTitle }} ({{ .Question.Difficulty }}){{ end }}
{{ .Question.GetFormattedContent }}
{{ .BlockCommentEnd }}
```

Just use multiple `lineComment` instead, then it will generate syntax error-free code

(maybe another TODO aside from adding language support)

Reply:
A better solution would be adding a `CommentLines(lines string) string` to Lang interface, and call it from template to comment lines. If a language doesn’t support multiline comments, CommentLines should fall back to use single line comments.

But this won’t at high priority given the low usage of Erlang.